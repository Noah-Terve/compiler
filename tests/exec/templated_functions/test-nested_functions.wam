template @l T0, T1 @r
T1 test2(T0 x, T1 y) {
    T0 w;
    T1 z;
    return y;
}

template @l T0, T1 @r
T0 test(T0 x, T1 y) {
    T0 w = x;
    T1 z = y;

    return test2 @l string, T0 @r ("hello", x);
}

print @l int @r(test @l int, int @r (123, 45));
print @l string @r(test @l string, bool @r ("home is where the heart is", true));
