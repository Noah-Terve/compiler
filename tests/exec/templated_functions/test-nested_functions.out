123home is where the heart is